{{cpp/title|generic_category}}
{{cpp/error/navbar}}
{{ddcl|header=system_error|since=c++11|
const std::error_category&amp; generic_category() noexcept;
}}

Obtains a reference to the static error category object for generic errors. The object is required to override the virtual function {{c|error_category::name()}} to return a pointer to the string {{c|"generic"}}. It is used to identify error conditions that correspond to the POSIX {{lc|errno}} codes.

===Parameters===
(none)

===Return value===
A reference to the static object of unspecified runtime type, derived from {{lc|std::error_category}}.

===Example===
{{example
|code=
#include &lt;cerrno&gt;
#include &lt;iostream&gt;
#include &lt;string&gt;
#include &lt;system_error&gt;

int main()
{
    std::error_condition econd = std::generic_category().default_error_condition(EDOM);
    std::cout &lt;&lt; "Category: " &lt;&lt; econd.category().name() &lt;&lt; '\n'
              &lt;&lt; "Value: " &lt;&lt; econd.value() &lt;&lt; '\n'
              &lt;&lt; "Message: " &lt;&lt; econd.message() &lt;&lt; '\n';
}
|output=
Category: generic
Value: 33
Message: Numerical argument out of domain
}}

===See also===
{{dsc begin}}
{{dsc inc|cpp/error/dsc system_category}}
{{dsc inc|cpp/error/dsc errc}}
{{dsc end}}

{{langlinks|de|es|fr|it|ja|pt|ru|zh}}